pkg/konnector/controllers/cluster/serviceexport/spec/spec_controller.go

```diff
@@ -104,25 +106,62 @@ func NewController(
 				return providerBindClient.KubeBindV1alpha1().APIServiceNamespaces(providerNamespace).Create(ctx, sn, metav1.CreateOptions{})
 			},
 			getProviderObject: func(ns, name string) (*unstructured.Unstructured, error) {
-				obj, err := providerDynamicInformer.Get(ns, name)
-				if err != nil {
-					return nil, err
+				if ns != "" {
+					obj, err := providerDynamicInformer.Get(ns, name)
+					if err != nil {
+						return nil, err
+					}
+					return obj.(*unstructured.Unstructured), nil
+				} else {
+					got, err := providerDynamicInformer.Get(ns, clusterscoped.Prepend(name, providerNamespace))
+					if err != nil {
+						return nil, err
+					}
+					obj := got.(*unstructured.Unstructured)
+					return clusterscoped.TranslateFromUpstream(obj.DeepCopy())
 				}
-				return obj.(*unstructured.Unstructured), nil
 			},
 			createProviderObject: func(ctx context.Context, obj *unstructured.Unstructured) (*unstructured.Unstructured, error) {
-				return providerClient.Resource(gvr).Namespace(obj.GetNamespace()).Create(ctx, obj, metav1.CreateOptions{})
+				if ns := obj.GetNamespace(); ns != "" {
+					return providerClient.Resource(gvr).Namespace(obj.GetNamespace()).Create(ctx, obj, metav1.CreateOptions{})
+				} else {
+					obj, err := clusterscoped.TranslateFromDownstream(obj, providerNamespace, providerNamespaceUID)
+					if err != nil {
+						return nil, err
+					}
+					if created, err := providerClient.Resource(gvr).Namespace(obj.GetNamespace()).Create(ctx, obj, metav1.CreateOptions{}); err != nil {
+						return nil, err
+					} else {
+						return clusterscoped.TranslateFromUpstream(created)
+					}
```

Comment:
```suggestion
					created, err := providerClient.Resource(gvr).Namespace(obj.GetNamespace()).Create(ctx, obj, metav1.CreateOptions{})
					if err != nil {
						return nil, err
					}
					return clusterscoped.TranslateFromUpstream(created)
```

Reply:
Thank you. I make a commit for the suggested changes.